ironfish/src/workerPool/worker.ts

```diff
@@ -151,6 +155,38 @@ function handleUnboxMessage({
   }
 }
 
+function handleMineHeader({
+  batchSize,
+  headerBytesWithoutRandomness,
+  initialRandomness,
+  miningRequestId,
+  targetValue,
+}: MineHeaderRequest): MineHeaderResponse {
+  const target = new Target(targetValue)
+  const randomnessBytes = new ArrayBuffer(8)
+
+  for (let i = 0; i < batchSize; i++) {
+    // The intention here is to wrap randomness between 0 inclusive and Number.MAX_SAFE_INTEGER inclusive
+    const randomness =
+      i > Number.MAX_SAFE_INTEGER - initialRandomness
+        ? i - (Number.MAX_SAFE_INTEGER - initialRandomness) - 1
+        : initialRandomness + i
+    new DataView(randomnessBytes).setFloat64(0, randomness, false)
+
+    const headerBytes = Buffer.concat([
+      Buffer.from(randomnessBytes),
+      headerBytesWithoutRandomness,
+    ])
+
+    const blockHash = hashBlockHeader(headerBytes)
+
+    if (Target.meets(new Target(blockHash).asBigInt(), target)) {
+      return { type: 'mineHeader', initialRandomness, miningRequestId, randomness }
+    }
+  }
+  return { type: 'mineHeader', initialRandomness }
+}
```

Comment:
Can we move this into the miner? And have the worker pool call it? I wanted to move the guts of these things into their respective modules but until we had something this chunky here it didn't seem too bad until now.

Reply:
👍 I'll move this